File: lecture/templates/lecture_template.java
import java.io.*;
import java.util.*;

class Solution {
    /**
     * M: number of rows in the lecture hall
     * N: number of columns in the lecture hall
     * K: number of friends who need seats
     * GRID: array of M strings of length N; '#' is a blocked seat, '-' is an empty seat
     *
     * Return a list of K int[2] {row, col} for the chosen seats.
     */
    static List<int[]> solve(int M, int N, int K, String[] GRID) {
        // YOUR CODE HERE
        List<int[]> result = new ArrayList<>();
        for (int i = 0; i < K; i++) {
            result.add(new int[]{0, 0});
        }
        return result;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
        StreamTokenizer in = new StreamTokenizer(br);
        in.resetSyntax();
        in.wordChars('!', '~');
        in.whitespaceChars(0, ' ');

        in.nextToken(); int T = Integer.parseInt(in.sval);
        for (int t = 0; t < T; t++) {
            in.nextToken(); int M = Integer.parseInt(in.sval);
            in.nextToken(); int N = Integer.parseInt(in.sval);
            in.nextToken(); int K = Integer.parseInt(in.sval);
            String[] GRID = new String[M];
            for (int i = 0; i < M; i++) {
                in.nextToken();
                GRID[i] = in.sval;
            }
            List<int[]> result = solve(M, N, K, GRID);
            for (int[] p : result) {
                pw.println(p[0] + " " + p[1]);
            }
        }
        pw.flush();
    }
}

File: lecture/templates/lecture_template.cpp
#include <bits/stdc++.h>
using namespace std;

/*
 * M: number of rows in the lecture hall
 * N: number of columns in the lecture hall
 * K: number of friends who need seats
 * GRID: vector of M strings of length N; '#' is a blocked seat, '-' is an empty seat
 *
 * Return a vector of K (row, col) pairs for the chosen seats.
 */
vector<pair<int, int>> solve(int M, int N, int K, vector<string>& GRID) {
    // YOUR CODE HERE
    return vector<pair<int, int>>(K, {0, 0});
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int T;
    cin >> T;
    while (T--) {
        int M, N, K;
        cin >> M >> N >> K;
        vector<string> GRID(M);
        for (int i = 0; i < M; i++) {
            cin >> GRID[i];
        }
        vector<pair<int, int>> result = solve(M, N, K, GRID);
        for (auto& p : result) {
            cout << p.first << " " << p.second << "\n";
        }
    }
    return 0;
}
